cmd/assert.go

```diff
@@ -64,81 +64,92 @@ var assertCmd = &cobra.Command{
-					if slos {
-						log.Logger.Info("SLOs are satisfied")
-					} else {
-						log.Logger.Info("SLOs are not satisfied")
-					}
-				} else if strings.HasPrefix(cond, slosByPrefix) {
-					version, err := exp.extractVersion(cond)
-					if err != nil {
-						os.Exit(1)
-					}
-					iv := exp.SLOsBy(version)
-					allGood = allGood && iv
-					if iv {
-						log.Logger.Info(version, " satisfies objectives")
-					} else {
-						log.Logger.Info(version, " does not satisfy objectives")
-					}
+		allGood, err := exp.Assert(conds, timeout)
+		if err != nil || !allGood {
+			os.Exit(1)
+		}
+	},
+}
+
+// Assert if experiment satisfies conditions
+func (exp *Experiment) Assert(conditions []string, to time.Duration) (bool, error) {
+	// check assert conditions
+	allGood := true
+	for {
+		for _, cond := range conditions {
+			if strings.ToLower(cond) == completed {
+				c := exp.completed()
+				allGood = allGood && c
+				if c {
+					log.Logger.Info("experiment completed")
 				} else {
-					log.Logger.Error("unsupported assert condition detected; ", cond)
-					os.Exit(1)
+					log.Logger.Info("experiment did not complete")
 				}
-			}
-			if allGood {
-				log.Logger.Info("all conditions were satisfied")
-				os.Exit(0)
-			} else {
-				if timeSpent > timeout {
-					log.Logger.Info("not all conditions were satisfied")
-					os.Exit(1)
+			} else if strings.ToLower(cond) == noFailure {
+				nf := exp.noFailure()
+				allGood = allGood && nf
+				if nf {
+					log.Logger.Info("experiment has no failure")
+				} else {
+					log.Logger.Info("experiment failed")
+				}
+			} else if strings.ToLower(cond) == slos {
+				slos := exp.SLOs()
+				allGood = allGood && slos
+				if slos {
+					log.Logger.Info("SLOs are satisfied")
 				} else {
-					time.Sleep(sleepTime)
-					timeSpent += sleepTime
+					log.Logger.Info("SLOs are not satisfied")
 				}
+			} else if strings.HasPrefix(cond, slosByPrefix) {
+				version, err := exp.extractVersion(cond)
+				if err != nil {
+					return false, err
+				}
+				iv := exp.SLOsBy(version)
+				allGood = allGood && iv
+				if iv {
+					log.Logger.Info(version, " satisfies objectives")
+				} else {
+					log.Logger.Info(version, " does not satisfy objectives")
+				}
+			} else {
+				log.Logger.Error("unsupported assert condition detected; ", cond)
+				return false, fmt.Errorf("unsupported assert condition detected; %v", cond)
 			}
 		}
-	},
+		if allGood {
+			log.Logger.Info("all conditions were satisfied")
+			return true, nil
+		} else {
+			if timeSpent > to {
+				log.Logger.Info("not all conditions were satisfied")
+				return false, nil
+			} else {
+				log.Logger.Info("sleeping %v ...", sleepTime)
+				time.Sleep(sleepTime)
+				timeSpent += sleepTime
+			}
+		}
+	}
 }
 
 // completed returns true if the experiment is complete
 // if the result stanza is missing, this function returns false
-func (exp *experiment) completed() bool {
+func (exp *Experiment) completed() bool {
```

Comment:
To implement the `get` function, we need to determine the values of conditions `completed` and `nofailure`. Currently, I am calling `Assert()` twice.  Perhaps, `complete()` and `noFailure()` could be exposed?  Alternatively, there might be a form of `Assert` that returns a list of booleans, one for each condition. We would still want the existing `Assert()` method as well -- it would wrap the new method.